### DISCUSSION TITLE
VNC Control Bar Removal

### DISCUSSION THREAD

**Discussion by fish-not-phish** [jlesage/docker-firefox#203] [Q&A] [answered]
When connecting to the container and entering the browser, I see there is a VNC Control Bar. Is it possible for me to remove this? I understand it's use case and functionality, but I my application of this container does not need it. If removal is not possible, would it be possible to change the settings that the user sees? Any advice or help would be greatly appreciated :)

**Comment by jlesage** [comment]
Currently there is no mechanism to remove it.
What settings are problematics to you ?  To better understand the problem and provide possible solutions, maybe you can describe your use case ?

**Reply by fish-not-phish** [reply]
I am using this docker container to give users an interactive browsing session through a web application. I am using an iFrame HTML element to display the interactive browser. However, the VBC Control Bar has some features that do not work in the environment that will require a webpage reload if they are clicked. For instance, if the "full screen" option is clicked from the VNC Control Bar, that webpage will throw an error because an iFrame cannot be made full screen.

**Comment by jlesage** [comment]
Since it doesn't work, the fullscreen button could be automatically removed when the page is loaded in a iFrame.  This change can be easily integrated in the image.

Is there another feature that doesn't work ?

**Reply by fish-not-phish** [reply]
I didn't even think of that yet. I guess I could use jQuery to remove elements within the iFrame once the webpage loads. That might work!

**Reply by jlesage** [reply, marked as answer]
I meant that I should do the fix in the Docker image itself.  So nothing would have to be done on your side :)

**Reply by fish-not-phish** [reply]
Thank you so much for the quick changes. I noticed you made a change to the base image, which is an earlier image layer. It still didn't work for me but I went ahead and made some adjustments in the ui.js and index.html to remove the functionality completely on my end. I definitely don't want to bother you with one persons requests! Amazing work by the way on these containers. I'm going to try and add support for some other browsers. If I'm able to successfully do that, I'll give you the supporting files so you can make your touches and publish them (if you'd like)!

**Comment by jlesage** [comment]
> It still didn't work for me

What do you mean exactly ?  Did you manually try the change I did ?

**Reply by fish-not-phish** [reply]
I manually downloaded the baseimage-gui repository locally, which has the UI.js and index.html files for the VNC Control Bar. Since the change you made didn't seem to work, I just commented out the fullscreen element within the HTML and commented out the supporting JS functions for fullscreen. I also commented out other settings I didn't want to be shown. I then pointed the firefox docker image to use the modified local base image. Sorry for the confusion.

**Reply by jlesage** [reply]
How did you test my change ?  I used https://iframetester.com to test: using Chrome, when the page is loaded in an iFrame, the fullscreen button does not appear.

**Reply by fish-not-phish** [reply]
I have a local project I am working on with a cross origin iFrame being used which is linked to the docker container through the src. I pulled your most recent docker image, spun up the container instance and it still had the fullscreen option inside of the VNC Control Bar.

**Reply by jlesage** [reply]
Ah ok.  Note that the change is not integrated yet into an official image.  If you want to test it, you can manually integrate it into a running container.

**Reply by Mincka** [reply]
Would it be possible to have more manual control over this Control Bar with environment variables?

I also have Kiosk scenarios where I don't want this control panel to be displayed, or maybe with a limited set of features (e.g. keep clipboard and remove full screen, or the opposite).

At least be able to disable it completely with an environment variable would be a nice feature to start.
